### PR TITLE
Make the dataset section dynamic for ReleaseInfo

### LINKED ISSUE #299: Create JSON Release (initial format)
The DP Release will be in JSON format and used to generate two formats:
 - HTML
 - PDF

**model**: ReleaseInfo
**note**: see field `dp_release` (JSON)
**updated format**:
```json
{
   "release_info":{
      "created":"2021-09-19T22:14:00.422715Z",
      "updated":"2021-09-19T22:14:00.422744Z",
      "object_id":"e1ba7dc6-ef98-4349-bae0-bb97126c82e4",
      "epsilon_used":1.0,
      "dp_release":{
         "name":"Replication Data for: Eye-typing experiment - step_700",
         "release_url":null,
         "created":{
            "iso":"2021-09-19T22:14:00.420117",
            "human_readable":"September 19, 2021 at 22:14:00:420117"
         },
         "application":"DP Creator",
         "application_url":"https://github.com/opendp/dpcreator",
         "differentially_private_library":{
            "name":"OpenDP",
            "url":"https://github.com/opendp/opendp",
            "version":"0.2.1"
         },
         "dataset":{
            "type":"dataverse",
            "name":"Replication Data for: Eye-typing experiment",
            "citation":null,
            "doi":"doi:10.7910/DVN/PUXVDH",
            "identifier":null,
            "release_deposit_info":{
               "deposited":false
            },
            "installation":{
               "name":"Mock Local Dataverse",
               "url":"http://127.0.0.1:8000/dv-mock-api"
            },
            "file_information":{
               "name":"Fatigue_data.tab",
               "identifier":null,
               "fileFormat":"text/tab-separated-values"
            }
         },
         "statistics":[
            {
               "statistic":"mean",
               "variable":"EyeHeight",
               "result":{
                  "value":-0.9602706204844246
               },
               "epsilon":0.5,
               "delta":0,
               "bounds":{
                  "min":-8.01,
                  "max":5.0
               },
               "missing_value_handling":{
                  "type":"insert_fixed",
                  "fixed_value":1.0
               },
               "confidence_interval":0.05,
               "accuracy":{
                  "value":0.42595057790021895,
                  "message":"Releasing mean for the variable EyeHeight. With at least probability 0.95 the output mean will differ from the true mean by at most 0.42595057790021895 units. Here the units are the same units the variable has in the dataset."
               }
            },
            {
               "statistic":"count",
               "variable":"TypingSpeed",
               "result":{
                  "value":183
               },
               "epsilon":0.5,
               "delta":0,
               "missing_value_handling":{
                  "type":"insert_fixed",
                  "fixed_value":"9"
               },
               "confidence_interval":0.05,
               "accuracy":{
                  "value":5.9914645483286035,
                  "message":"Releasing count for the variable TypingSpeed. With at least probability 0.95 the output count will differ from the true count by at most 5.9914645483286035 units. Here the units are the same units the variable has in the dataset."
               }
            }
         ]
      }
   }
}
```

---

Related: #21, #28, #31

### PR DESCRIPTION
**Pull request by raprasad**
- Note: Several differences in the final release based on field availability from Dataverse metadata...
  - Examples here: https://github.com/opendp/dpcreator/issues/299#issuecomment-922545079
- Also, main format may have changes, updated in the top issue description:
  - https://github.com/opendp/dpcreator/issues/299

### CODE REVIEW

File: server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (ecowan) [inline server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json]: Should all of these json files be included in git?

File: server/opendp_apps/analysis/release_info_formatter.py [hunk 2]
```diff
@@ -41,15 +53,32 @@ def get_readable_datetime(self, dt_obj: dt):
...
+        if ds_formatter.has_error():
+            self.add_err_msg(ds_formatter.get_err_msg())
+            return
+        else:
+            dataset_dict = ds_formatter.get_formatted_info()
+
+        self.release_dict = {
             "name": str(self.release_util.analysis_plan),
-            "release_url": '(not available, should be url within DP Creator)',
+            "release_url": None,    # to do...
```
Comment (ecowan) [inline server/opendp_apps/analysis/release_info_formatter.py]: What do we need to do for the release URL? Should we do it before merging this?

**Review by ecowan — COMMENTED** [review]
I left two questions, look good otherwise

File: server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json (hunk 1 above)
Comment (raprasad) [reply server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json]: Thanks. The .gitignore was updated but not the actual files

File: server/opendp_apps/analysis/release_info_formatter.py (hunk 2 above)
Comment (raprasad) [reply server/opendp_apps/analysis/release_info_formatter.py]: I moved it to this issue #34 and put comments in the code linking to the issue.
e.g. Probably enough for a separate ticket/endpoint. Is it a "public" detail page? File/file-like objects aren't handled through regular MEDIA_URL, etc and need some extra code

File: server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json (hunk 1 above)
Comment (raprasad) [reply server/release-files/2021/09/19/release-0096c2e0-2c55-41c5-8ba1-896f0b072f0a.json]: (files now deleted)

**Pull request merged**